tgui/packages/tgui/interfaces/Keybinds/Keybind.tsx

```diff
@@ -19,22 +20,44 @@ interface KeybindProps {
 export const Keybind = (props: KeybindProps) => {
   const { act } = useBackend<KeybindsData>();
   const { keybind, isFocused, setFocusedKey } = props;
+
+  const [lastEvent, setLastEvent] =
+    useState<React.KeyboardEvent<HTMLDivElement>>();
+
   return (
     <LabeledList.Item label={keybind.label}>
       <Button
         onKeyDown={(event) => {
           event.preventDefault();
+          if (!isFocused) {
+            return;
+          }
           if (isEscape(event.key)) {
             setFocusedKey('');
             return;
           }
           if (!isStandardKey(event)) {
+            // Store the event in case the key is released in which case we'll want to bind to this non-Standard key.
+            setLastEvent(event);
```

Comment:
Be aware that state setting causes re-renders. Is that acceptable here, or would this being in a `useRef` and having its state managed more directly be more appropriate?